stashcp/__init__.py

```diff
@@ -325,36 +325,52 @@ def download_xrootd(sourceFile, destination, debug, payload):
     xrd_exit = timed_transfer(filename=sourceFile, debug=debug, cache=cache, destination=destination)
 
     payload['xrdexit1']=xrd_exit
+    tries = 1
+    payload['cache1'] = cache
 
     if xrd_exit=='0': #worked first try
         logging.debug("Transfer success using %s", nearest_cache)
-        payload['tries'] = 1
+        status = "First Cache Success"
         payload['cache'] = cache
 
-    else: #pull from origin
-        logging.info("XrdCP from cache failed on %s, pulling from main redirector", nearest_cache)
+    if xrd_exit != '0': # pull from second nearest cache
+        cache = nearest_cache_list[1]
```

Comment:
This assumes that there are at least 2 caches in nearest_cache_list, which may not be true if using a custom caches.json file.

Reply:
Good point.  Changed in f7df5a4